fbw-common/src/wasm/systems/systems/src/enhanced_gpwc/mod.rs

```diff
@@ -0,0 +1,184 @@
+        }
+    }
+
+    fn update_position_data(&mut self, adirs_output: &impl AdirsMeasurementOutputs) {
+        /*
+         * documentation hints:
+         *   - EGPWC has direct connection to GPS sensor && ADIRS_1
+         *   - uses direct GPS data if ADIRS_1 is unavailable
+         * TODO:
+         *   - implement logic as soon as GPS sensor is available
+         */
+        self.latitude = adirs_output.latitude(1);
+        self.longitude = adirs_output.longitude(1);
+        self.altitude = adirs_output.altitude(1);
+        self.heading = adirs_output.heading(1);
+        self.vertical_speed = adirs_output.vertical_speed(1);
+    }
+
+    pub fn update(
+        &mut self,
+        adirs_output: &impl AdirsMeasurementOutputs,
+        lgcius: &LandingGearControlInterfaceUnitSet,
+    ) {
+        self.update_position_data(adirs_output);
+        self.gear_is_down = lgcius.lgciu1().main_down_and_locked();
+
+        self.navigation_displays.iter_mut().for_each(|display| {
+            display.update(
+                &self.navigation_display_range_lookup,
+                adirs_output.is_fully_aligned(1),
+            )
+        });
+    }
+}
+
+impl SimulationElement for EnhancedGPWC {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_powered = buses.is_powered(self.powered_by)
+    }
+
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        let destination_long: f64 = reader.read(&self.fm1_destination_longitude_id);
+        let destination_lat: f64 = reader.read(&self.fm1_destination_latitude_id);
+        let destination_long_ssm: u32 = reader.read(&self.fm1_destination_longitude_ssm_id);
+        let destination_lat_ssm: u32 = reader.read(&self.fm1_destination_latitude_ssm_id);
+
+        self.destination_longitude = Arinc429Word::new(
+            Angle::new::<degree>(destination_long),
+            SignStatus::from(destination_long_ssm),
+        );
+        self.destination_latitude = Arinc429Word::new(
+            Angle::new::<degree>(destination_lat),
+            SignStatus::from(destination_lat_ssm),
+        );
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write_arinc429(
+            &self.egpwc_destination_longitude_id,
+            self.destination_longitude.value().get::<degree>(),
+            self.destination_longitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_destination_latitude_id,
+            self.destination_latitude.value().get::<degree>(),
+            self.destination_latitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_present_latitude_id,
+            self.latitude.value().get::<degree>(),
+            self.latitude.ssm(),
+        );
+        writer.write_arinc429(
+            &self.egpwc_present_longitude_id,
+            self.longitude.value().get::<degree>(),
+            self.longitude.ssm(),
+        );
```

Comment:
I think it would be a good idea to add the following to the `read_write_uom` macro so that you can just use `writer.write`.
```rust
impl<T: Writer> Write<Arinc429Word<$t>> for T {
    fn convert(&mut self, value: Arinc429Word<$t>) -> f64 {
        let v = self.convert(value.value());
        Arinc429Word::new(v, value.ssm()).into()
    }
}
```
(If you don't know how I can do it for you and commit it to this PR)